Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -1338,6 +1338,8 @@ void WebPageProxy::initializeWebPage()
     setDrawingArea(pageClient().createDrawingAreaProxy());
     ASSERT(m_drawingArea);
 
+    websiteDataStore().didInitialize(*this);
```

Comment:
There is another `initializeWebPage` in `ProvisionalPageProxy`, see comment in `WebPageProxy::finishAttachingToWebProcess`. 
We probably should add and remove page to/from `WebsiteDataStore` in `pageBeginUsingWebsiteDataStore()` and `pageEndUsingWebsiteDataStore()`.

Reply:
Oh, nice — `pageBeginUsingWebsiteDataStore` and `pageEndUsingWebsiteDataStore` seem perfect for my use case. If it's reasonable, I'll make it take a `WebPageProxy&` as well, so that we can just add it to the set.